### PR TITLE
[Tests-Only]Add UI tests for dot in folder name

### PR DESCRIPTION
**Pull request by sakshamgurung**
## Description

- Add UI tests for dot in folder name
## Related Issue

- Test for https://github.com/owncloud/ocis/issues/2193

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUICreateFilesFolders/createFolders.feature [hunk 1]
```diff
@@ -67,3 +67,27 @@ Feature: create folders
...
+      | fo.1     |
+      | fo...1.. |
+      | ...      |
+      | ..fo     |
+      | fo.xyz   |
+
+
+#    Scenario: Try to create a folder with .part in name
+#      When the user tries to create a folder with the invalid name "folder.part" using the webUI
+#      Then the create folder button should be disabled
```
Comment (kiranparajuli589) [inline tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:93]: left overs?

Comment (sakshamgurung) [reply tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:93]: I try I fail. Any suggestion.

Comment (kiranparajuli589) [reply tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:93]: why does the test fail? there might be an issue. please look at the failing causes more carefully.

File: tests/acceptance/features/webUIDeleteFilesFolders/deleteFilesFolders.feature [hunk 2]
```diff
@@ -323,3 +323,39 @@ Feature: deleting files and folders
...
+      | fo.xyz   |
+    And the user has reloaded the current page of the webUI
+    And as "Alice" these folders should be listed on the webUI
+      | folders  |
+      | fo.      |
+      | fo.1     |
+      | fo...1.. |
+      | ...      |
+      | ..fo     |
+      | fo.xyz   |
```
Comment (jasson99) [inline tests/acceptance/features/webUIDeleteFilesFolders/deleteFilesFolders.feature:345]: we don't need this here.

File: tests/acceptance/features/webUITrashbinDelete/trashbinDelete.feature [hunk 3]
```diff
@@ -120,3 +120,33 @@ Feature: files and folders can be deleted from the trashbin
...
+      | fo.xyz   |
+    And the following folders have been deleted by user "Alice"
+      | name     |
+      | fo.      |
+      | fo.1     |
+      | fo...1.. |
+      | ...      |
+      | ..fo     |
+      | fo.xyz   |
+    And the user reloads the current page of the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUITrashbinDelete/trashbinDelete.feature:142]:
```suggestion
    And the user has reloaded the current page of the webUI
```

File: tests/acceptance/features/webUITrashbinRestore/trashbinRestore.feature [hunk 4]
```diff
@@ -274,3 +274,48 @@ Feature: Restore deleted files/folders
...
+      | ..fo     |
+      | fo.xyz   |
+    And the user batch restores the marked files using the webUI
+    And the user reloads the current page of the webUI
+    Then folder "fo." should not be listed on the webUI
+    And folder "fo.1" should not be listed on the webUI
+    And folder "fo...1.." should not be listed on the webUI
+    And folder "..." should not be listed on the webUI
+    And folder "..fo" should not be listed on the webUI
+    And folder "fo.xyz" should not be listed on the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUITrashbinRestore/trashbinRestore.feature:312]: use data table

File: tests/acceptance/features/webUITrashbinRestore/trashbinRestore.feature [hunk 5]
```diff
@@ -274,3 +274,48 @@ Feature: Restore deleted files/folders
...
+    When the user browses to the files page
+    #sometimes the below test passes sometimes below folder(not always same folder) doesn't exist
+    Then folder "fo." should be listed on the webUI
+    And folder "fo.1" should be listed on the webUI
+    And folder "fo...1.." should be listed on the webUI
+    And folder "..." should be listed on the webUI
+    And folder "..fo" should be listed on the webUI
+    And folder "fo.xyz" should be listed on the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUITrashbinRestore/trashbinRestore.feature:320]: use data table

File: tests/acceptance/features/webUICreateFilesFolders/createFolders.feature (hunk 1 above)
Comment (sakshamgurung) [reply tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:93]: https://github.com/owncloud/web/issues/5518

Comment (kiranparajuli589) [reply tests/acceptance/features/webUICreateFilesFolders/createFolders.feature]: You can convert this into a bug demostration... Tag the scenario with the created issue

File: tests/acceptance/features/webUITrashbinDelete/trashbinDelete.feature [hunk 6]
```diff
@@ -120,3 +120,33 @@ Feature: files and folders can be deleted from the trashbin
     When the user clears the trashbin
     Then the success message with header "All deleted files were removed" should be displayed on the webUI
     And there should be no resources listed on the webUI
+
+
+  Scenario: Delete folders with dot in the name from trashbin
```
[the remark this answers is not in the document]
Comment (sakshamgurung) [reply tests/acceptance/features/webUITrashbinDelete/trashbinDelete.feature:125]: Sorry but I don't understand. Should I change scenario title or what?

[the remark this answers is not in the document]
Comment (SwikritiT) [reply tests/acceptance/features/webUITrashbinDelete/trashbinDelete.feature:125]: I think you should add a step ` When the user browses to the trashbin page`. The current scenario doesn't delete from trashbin.

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 7]
```diff
@@ -989,6 +989,13 @@ Given('user {string} has created the following files', async function(userId, en
   return client
 })
 
+When('the user/public creates the following folders using the webUI', async function(entryList) {
+  for (const entry of entryList.rows()) {
+    await client.page.personalPage().createFolder(entry[0])
+  }
+  return client
```
Comment (saw-jan) [inline tests/acceptance/stepDefinitions/filesContext.js:996]: you have sync function so no need
```suggestion
```

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/17408/38/1
The following scenarios passed on retry:
- webUISharingExternal/federationSharing.feature:364

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by swoichha — COMMENTED** [review]
looks good :+1:

File: tests/acceptance/features/webUICreateFilesFolders/createFolders.feature [hunk 8]
```diff
@@ -67,3 +67,28 @@ Feature: create folders
...
+      | folders  |
+      | fo.      |
+      | fo.1     |
+      | fo...1.. |
+      | ...      |
+      | ..fo     |
+      | fo.xyz   |
+
+
+  @skipOnOC10 @issue-5518
```
Comment (phil-davis) [inline tests/acceptance/features/webUICreateFilesFolders/createFolders.feature:91]: Note: On oC10, `.part` files are not allowed. So this scenario will always be tagged `skipOnOC10`

The issue is about improving the current behavior of `web` with oC10. It should give the user some good feedback about why the resource cannot be created. When that is improved, then an extra scenario can be added here that demonstrates the behavior with oC10 (and will be skipped on OCIS)

**Review by phil-davis — APPROVED** [review]

**Pull request merged**